pgjdbc/src/test/java/org/postgresql/test/udt/CustomTypePriorityTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2018, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.test.udt;
+
+import org.postgresql.core.BaseConnection;
+import org.postgresql.jdbc.PreferQueryMode;
+import org.postgresql.test.TestUtil;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.sql.Types;
+import java.util.Collections;
+import java.util.Map;
+
+public class CustomTypePriorityTest {
```

Comment:
Just a random thought
1) you might want to use `extends BaseTest4` (to save `setUp` `tearDown`) repetitions.
2) Please add newly added test files to an existing suite. Otherwise the test files are not run in the TravisCI

Reply:
Thanks.  We had used `BaseTest4` before, we we're sure we should.

We'll include all these tests at the end of `Jdbc2TestSuite`.

Reply:
Thanks done.  I can confirm at least some of our tests were running, by the way:

https://travis-ci.org/pgjdbc/pgjdbc/builds/472061741?utm_source=github_status&utm_medium=notification
https://travis-ci.org/pgjdbc/pgjdbc/jobs/472061755

Are you sure we cannot have a different test suite?  It sure is convenient to just fire of this suite while developing.